helper.sh

```diff
@@ -89,6 +93,8 @@ get_type3() {
     return 0
 }
 
+# like get_type, but use full componant name, instead of API function Name
```

Comment:
Retrieve the type directly from the full component JSON rather than using the API function name in get_type.